### ISSUE TITLE
Add Support for RIFE

### ISSUE THREAD

**Issue by Tsubajashi** [Mar2ck/DAIN-Vulkan-GUI#6]
Hi!
at first, thanks for the great CLI, its pretty nice to use.
secondly, there is a new interpolation method in town - https://github.com/hzwer/arXiv2020-RIFE
would it be possible for you to integrate that one too? because its fast even on cpu, and doesnt seem to look bad at all.

Thanks in advance

**Comment by Mar2ck** [comment]
I'm glad you like it! I definitely can add RIFE support and have already been considering it. Hopefully there'll be a vulkan port at some point in the future but for now performance is good enough that running it on CPU is a viable option. I have had trouble getting it to run in cpu mode though so I'll look into it more.

**Comment by Tsubajashi** [comment]
i personally was only really able to get it to run in google collab, since they put the link in there.
either my pip environment is just plain broken, or im just too tired

**Comment by Mar2ck** [comment]
I just got it to run, pytorch has a seperate version just for cpu-only usage which is slightly annoying since it means i can't easily switch from cpu to cuda but that's not a huge deal. Performance is fairly good considering, Rife on cpu (R7 3700x) is about 6x faster then Dain-ncnn on gpu (Gtx 1660 ti) but with much better results visually

**Comment by Tsubajashi** [comment]
That sounds great!
i thought about pytorch, but the reason why i think my environment is broken, is it cannot find "cv2", which i obviously have everywhere it searches for. which sounds just too odd for me. its been a while where i last saw python.

**Comment by Mar2ck** [comment]
cv2 should be from opencv-python. If it helps i used anaconda with these to get it working:
```
conda create -n rife-cpu
conda activate rife-cpu
conda install python=3.8
conda install pytorch torchvision torchaudio cpuonly -c pytorch
conda install pip numpy
python -m pip install tqdm opencv-python
```

**Comment by Tsubajashi** [comment]
seems to work. Thanks!

**Comment by Tsubajashi** [comment]
FYI: the author improved the speed and model a lot. did you have another look into it?

**Comment by Mar2ck** [comment]
Speed is improved with the parallelization and the new model fixed the weird wobbling artifacting. I'm currently figuring out how to write a wrapper for RIFE since the code is unmodular but it's not too complicated
| v1.1 | v1.2| 
| --- | --- |
| ![old model](https://user-images.githubusercontent.com/6759254/99829900-d3777280-2b54-11eb-9203-529df20da76e.png) | ![new model](https://user-images.githubusercontent.com/6759254/99829934-dffbcb00-2b54-11eb-97b2-99a4754d4ab8.png)

**Comment by Mar2ck** [comment]
I added support to the git verison, use `-e rife`. I'll put out a release with rife support at some point

**Comment by Tsubajashi** [comment]
I‘ll test it in a day, thanks!

LG
Jan Jonas Willumeit
________________________________

**Comment by Tsubajashi** [comment]
Since a few minutes, rife-ncnn-vulkan exists and seems to work.
can you implement that one?

**Comment by Mar2ck** [comment]
Yes i will add support for the ncnn version soon. The original version of rife would've inflated the release size massively since PyTorch is almost 1gb on its own

**Comment by Tsubajashi** [comment]
yea, pytorch is something which isnt really nice to pakage, thats true

**Comment by Tsubajashi** [comment]
im not entirely sure how i would run the git version, since i saw you made a commit with the rife definitions, and wanted to test it.

**Comment by Mar2ck** [comment]
I haven't fully added the functionality yet. I'll update the release and google colab after i'm done so you can use it

**Comment by Tsubajashi** [comment]
i see, Thank you! :)

**Comment by Mar2ck** [comment]
Just released 0.0.5 builds with rife-ncnn-vulkan support

**Comment by Tsubajashi** [comment]
Works perfectly, thanks! :)
